### PR TITLE
JDK-8301998: Update HarfBuzz to 7.0.1

### PR DESCRIPTION
**Pull request by honkar-jdk**
HarfBuzz library updated from v4.4.1 to v7.0.1

- harfbuzz.md file updated
- Added -DHB_NO_VISIBILITY to HARFBUZZ_CFLAGS in Awt2DLibraries.gmk to avoid build issues on linux and macos targets.
- GPOS.hh moved to Layout to Layout/GPOS to match upstream changes.
- hb-ot-color-* files moved to OT/Color to match upstream changes.
- 41 newly added, 8 deleted and 224 modified files (src + harfbuzz.md).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301998](https://bugs.openjdk.org/browse/JDK-8301998): Update HarfBuzz to 7.0.1

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12913/head:pull/12913` \
`$ git checkout pull/12913`

Update a local copy of the PR: \
`$ git checkout pull/12913` \
`$ git pull https://git.openjdk.org/jdk pull/12913/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12913`

View PR using the GUI difftool: \
`$ git pr show -t 12913`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12913.diff">https://git.openjdk.org/jdk/pull/12913.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
Bulid changes look good.

**Comment by jcking** [comment]
FYI HarfBuzz most recent is 7.1.0 as of last week. Might make sense to just bite the bullet as it were and to 7.1.0 instead of 7.0.1.

**Comment by prrace** [comment]
> FYI HarfBuzz most recent is 7.1.0 as of last week. Might make sense to just bite the bullet as it were and to 7.1.0 instead of 7.0.1.

We already switched once from 7.0 to 7.0.1. By the time we've got 7.1 in review, there might be a 7.2 ..
I think it best to finish this big upgrade which is 99% done rather than starting over with an even bigger upgrade. Going from 7.0.1 to (eg) 7.1 or a hypothetical 7.2 will then be easier.

**Comment by honkar-jdk** [comment]
Builds and Automated tests are successful with the new Harfbuzz changes.
Required manual testing was carried out on Mac, Windows & Linux (Ubuntu) and works as expected.

File: src/java.desktop/share/native/libharfbuzz/hb.hh [hunk 1]
```diff
@@ -103,20 +103,20 @@
 #pragma GCC diagnostic warning "-Wdisabled-optimization"
 #pragma GCC diagnostic warning "-Wdouble-promotion"
 #pragma GCC diagnostic warning "-Wformat=2"
+#pragma GCC diagnostic warning "-Wformat-signedness"
 #pragma GCC diagnostic warning "-Wignored-pragma-optimize"
```
Comment (prrace) [inline src/java.desktop/share/native/libharfbuzz/hb.hh:107]: Reading these changes makes me think we should re-check if all the DISABLED_WARNINGS we have for
harbuzz in Awt2DLibraries.gmk (see lines 466-470 in that file) are still needed.
In particular for gcc + clang (ie windows and mac builds).
Can you comment them out (in that file) and test the build ? Any that are now un-needed can be removed.

Comment (honkar-jdk) [reply src/java.desktop/share/native/libharfbuzz/hb.hh:107]: @prrace Sure Phil. I'll comment it out and check if there are any un-needed flags in DISABLED_WARNINGS section.

Comment (honkar-jdk) [reply src/java.desktop/share/native/libharfbuzz/hb.hh:107]: @prrace Checked by commenting out the HARFBUZZ_DISABLED_WARNINGS section in Awt2DLibraries.gmk, all the currently listed warnings (for specific platform compilers) are required else the build fails.

**Review by prrace — APPROVED** [review]
Ok - we are good to go.

**Pull request closed without merging**